apps/common-app/src/examples/RuntimeTests/tests/layoutAnimations/layout/testComponents.test.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState, useEffect } from 'react';
+import { View, StyleSheet } from 'react-native';
+import Animated from 'react-native-reanimated';
+import { useTestRef } from '../../../ReanimatedRuntimeTestsRunner/RuntimeTestsApi';
+
+export const TRANSITION_REF = 'TRANSITION_REF';
+export enum Direction {
+  UP = 'UP',
+  DOWN = 'DOWN',
+  LEFT = 'LEFT',
+  RIGHT = 'RIGHT',
+  RIGHT_UP = 'RIGHT_UP',
+  LEFT_UP = 'LEFT_UP',
+}
+
+export const TransitionUpOrDown = ({ layout, direction }: { layout: any; direction: Direction }) => {
+  const [show, setShow] = useState(true);
+  const ref = useTestRef(TRANSITION_REF);
+
+  useEffect(() => {
+    setShow(false);
+  }, [setShow]);
+
+  return (
+    <View style={styles.containerVertical}>
+      {direction === Direction.UP && (
+        <>
+          {show && <Animated.View layout={layout} style={styles.animatedBox} />}
+          <Animated.View layout={layout} style={styles.animatedBox} />
+          <Animated.View ref={ref} layout={layout} style={[styles.animatedBox, styles.mainBox]} />
+        </>
+      )}
+      {direction === Direction.DOWN && (
```

Comment:
What does the `direction` property describe?

Reply:
It is the direction of movement of the component that we record a snapshot of.

Reply:
Please see that the orange view is moving in different directions:

https://github.com/software-mansion/react-native-reanimated/assets/56199675/58c19501-f6dc-47c3-800c-4c5cd4ddb4c6